model_analyzer/triton/server/server_docker.py

```diff
@@ -69,16 +71,39 @@ def start(self):
                 docker.types.DeviceRequest(device_ids=self._gpus,
                                            capabilities=[['gpu']])
             ]
-        environment = {
-            'CUDA_VISIBLE_DEVICES': ','.join([uuid for uuid in self._gpus])
-        }
 
-        # Mount required directories
-        volumes = {
-            self._server_config['model-repository']: {
-                'bind': self._server_config['model-repository'],
-                'mode': 'ro'
+        # Set environment inside container.
+        # Supports only strings, and value lookups/concats
+        env_cmds = []
+        if env:
+            env.update({
+                'CUDA_VISIBLE_DEVICES':
+                ','.join([uuid for uuid in self._gpus])
+            })
+
+            # Filter out variables that need to be set within container
+            # If the variable is LD_PRELOAD, it cannot be set in docker env
```

Comment:
Is `LD_PRELOAD` the only variable that needs to be handled in this way? Can we be flexible to support both modes for all the variables? For example, `X=$VALUE` indicates that it should be set from the outside container and `X=$$VALUE` indicates that it should be set from inside the container.

Reply:
Yes, LD_PRELOAD seems to need to be set immediately before the program being run, and its best to set it only for that program. 

Regards to your suggestion, I am wondering which case is not covered in the code as is? If the variables value contains a lookup it will need to be set from inside the container. 

I suppose there are cases where the you may need to set the variable to a value containing "$" is that the case you are referring to?

Reply:
Given that the name of the option is "triton_server_environment" why not set all the parameters like LD_PRELOAD?

> I suppose there are cases where the you may need to set the variable to a value containing "$" is that the case you are referring to?

Right, I was wondering which environment will be used for the resolution of that env variable? Is it going to use the environment variables inside the container or it is going to use the outside environment to resolve that variable? For example, if I set "X=$PATH", is going to be resolved to the PATH variable inside the container or outside the container.

Reply:
That will be resolved inside the container. Anything with a $ will be resolved inside. The reason i opted to split some up was in case the command needs to be debugged, i didn't want the tritonserver command to be inundated with environment variables. If you think this is unnecessary I can set them all the same way.

Reply:
I think it is a bit strange to single out `LD_PRELOAD` as a special environment variable. Can you elaborate on what you mean by the command needs to be debugged?

Reply:
The tritonserver command that ultimately gets run inside the container will be somethign like `X=1 Y=2 tritonserver ....` If we have a lot of environment variables, i think this will get messy. This was my only concern. Otherwise i think you're probably right that we're better off setting all the variables in the command.

Reply:
I will make this change and rerun the CI.